include/deal.II/matrix_free/hanging_nodes_internal.h

```diff
@@ -376,6 +376,21 @@ namespace internal
     inline void
     HangingNodes<3>::setup_line_to_cell(const Triangulation<3> &triangulation)
     {
+      // In case all our cells are on the same level, there is nothing to
+      // do. We only need to check for active cells (non-artificial) that are
+      // on a level coarser than the finest one.
+      bool all_cells_on_same_level = true;
+      for (const auto &cell : triangulation.active_cell_iterators())
+        if (cell->level() == static_cast<int>(triangulation.n_levels()) - 1)
+          break;
```

Comment:
Yes, I do: I do not want to loop through all cells, since it is enough to complete the loop through all active cells on a coarser level and see if there is one non-artificial. We maybe have an iterator filter, but after looking for 30 seconds I decided I was quicker to implement it. Do you want me to not assume the order of active cells level by level? I could do so by traversing all cells and removing this `if` clause here, at a probably negligible impact on performance.